### PR TITLE
Improve binding error

### PR DESCRIPTION
**Pull request by eliemichel**
**Description**
Error messages in case of a mismatch between resource binding as it is in the shader and as it is in the binding layout only say "there is a mismatch" without precising what binding type was found where. This improves the error message.

**Checklist**

- [x] Run `cargo fmt`.
- [x] Run `taplo format`.
- [x] Run `cargo clippy`. If applicable, add:
  - [x] `--target wasm32-unknown-unknown`
  - [x] `--target wasm32-unknown-emscripten`
- [x] Run `cargo xtask test` to run tests.
- [x] Add change to `CHANGELOG.md`. See simple instructions inside file.

### CODE REVIEW

File: wgpu-core/src/validation.rs [hunk 1]
```diff
@@ -20,6 +20,33 @@ enum ResourceType {
...
 
+#[derive(Clone, Debug)]
+pub enum BindingTypeName {
+    Buffer,
+    Texture,
+    Sampler,
+    AccelerationStructure,
+}
+
+fn map_resource_to_type_name(ty: &ResourceType) -> BindingTypeName {
```
Comment (cwfitzgerald) [inline wgpu-core/src/validation.rs]: I would expect this to be an inherent function on BindingTypeName, like `BindingTypeName::from_resource_type`

Comment (cwfitzgerald) [reply wgpu-core/src/validation.rs]: Actually I think an actual `From` impl will probably be best, then the call sites can just use `into`.

**Review by cwfitzgerald — DISMISSED** [review]

File: wgpu-core/src/validation.rs (hunk 1 above)
Comment (eliemichel) [reply wgpu-core/src/validation.rs]: Will do! I admit I was really not sure where to put this, not 100% familiar with rust idioms, thanks for the feedback. :)

**Comment by ErichDonGubler** [comment]
Gimme a cool `CHANGELOG` entry, and I'll merge it! 🙂

**Review by ErichDonGubler — APPROVED** [review]

**Comment by eliemichel** [comment]
Here we are :)

**Review by cwfitzgerald — COMMENTED** [review]
Noice

**Review by cwfitzgerald — APPROVED** [review]

**Pull request merged**